src/winmain.cpp

```diff
@@ -917,6 +917,47 @@ bool runInstaller(const wstring& app2runPath, const wstring& binWindowsClassName
 	return true;
 }
 
+// Returns a folder suitable for exe installer download destination.
+std::wstring getDestDir()
+{
+	// Some users might unset the temp variable, so need to account for that.
+	// Other fallbacks: %AppData%, %UserProfile% or GetModuleFileName().
+	size_t requiredSize = 0;
+	std::vector<wchar_t> buffer(MAX_PATH + 1, 0);
+	const wchar_t* envList[] = { L"TEMP", L"TMP", L"UserProfile", L"LocalAppData", L"ProgramData"};

```

Comment:
Ok, I'll fix it.

Reply:
@donho How about one more fallback option - %UserProfile%\Downloads?
It can be obtained using [SHGetKnownFolderPath](https://docs.microsoft.com/en-us/windows/win32/api/shlobj_core/nf-shlobj_core-shgetknownfolderpath?redirectedfrom=MSDN) function with the argument [FOLDERID_Downloads](https://docs.microsoft.com/en-us/windows/win32/shell/knownfolderid)

Reply:
@mere-human 
> How about one more fallback option - %UserProfile%\Downloads?

OK for me.

Reply:
Done.